src/ooxml/java/org/apache/poi/xssf/usermodel/helpers/XSSFColumnShifter.java

```diff
@@ -0,0 +1,170 @@
+        {    
+            Row row = shiftingSheet.getRow(rowNo);
+            if(row == null)
+                continue;
+            for (int columnIndex = lastShiftColumnIndex; columnIndex >= firstShiftColumnIndex; columnIndex--){ // process cells backwards, because of shifting 
+                XSSFCell oldCell = (XSSFCell)row.getCell(columnIndex);
+                Cell newCell = null;
+                if(oldCell == null){
+                    newCell = row.getCell(columnIndex + shiftStep);
+                    newCell = null;
+                    continue;
+                }
+                else {
+                    newCell = row.createCell(columnIndex + shiftStep, oldCell.getCellTypeEnum());
+                    cloneCellValue(oldCell,newCell);
+                    if(columnIndex <= firstShiftColumnIndex + shiftStep - 1){ // clear existing cells on place of insertion
+                        oldCell.setCellValue("");
+                        oldCell.setCellType(CellType.STRING);
+                    }
+                }
+            }
+        }
+    }
+    private void shiftColumnsLeft(){
+        for(int rowNo = 0; rowNo <= shiftingSheet.getLastRowNum(); rowNo++)
+        {    
+            XSSFRow row = (XSSFRow)shiftingSheet.getRow(rowNo);
+            if(row == null)
+                continue;
+            for (int columnIndex = 0; columnIndex < row.getLastCellNum(); columnIndex++){ 
+                XSSFCell oldCell = (XSSFCell)row.getCell(columnIndex);
+                if(columnIndex >= firstShiftColumnIndex + shiftStep && columnIndex < row.getLastCellNum() - shiftStep){ // shift existing cell 
+                    org.apache.poi.ss.usermodel.Cell newCell = null;
+                    newCell = row.getCell(columnIndex - shiftStep);
+                    if(oldCell != null){
+                        if(newCell != null){
+                            oldCell.setCellType(newCell.getCellType());
+                            cloneCellValue(newCell, oldCell);
+                        }
+                        else {
+                            oldCell.setCellType(CellType.STRING);
+                            oldCell.setCellValue("");
+                        }
+                    }
+                    else {
+                        oldCell = row.createCell(columnIndex);
+                        if(newCell != null){
+                            oldCell.setCellType(newCell.getCellType());
+                            cloneCellValue(newCell, oldCell);
+                        }
+                        else {
+                            oldCell.setCellType(CellType.STRING);
+                            oldCell.setCellValue("");
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    public void shiftComments(XSSFVMLDrawing vml, int startColumnIndex, int endColumnIndex, final int n, CommentsTable sheetComments){
```

Comment:
`XSSFVMLDrawing` isn't in the signature of most public POI methods. Is it needed here, or can you determine what `vml` is inside this method?

Can `shiftComments` be `private`, `package-private`, or `protected`?